src/EnergyPlus/SurfaceGeometry.cc

```diff
@@ -2757,6 +2809,65 @@ namespace SurfaceGeometry {
         }
     }
 
+    void CreateMissingSpaces(EnergyPlusData &state, bool &ErrorsFound)
+    {
+        static constexpr std::string_view RoutineName("CreateMissingSpaces: ");
+        // Scan surfaces to see if Space was assigned in input
+        for (int surfNum = 1; surfNum <= state.dataSurface->TotSurfaces; ++surfNum) {
+            auto &thisSurf = state.dataSurface->Surface(surfNum);
+            if (!thisSurf.HeatTransSurf) continue; // ignore shading surfaces
+            if (thisSurf.spaceNum > 0) {
+                state.dataHeatBal->Zone(thisSurf.Zone).anySurfacesWithSpace = true;
+            } else {
+                state.dataHeatBal->Zone(thisSurf.Zone).anySurfacesWithoutSpace = true;
+            }
+        }
+
+        // Create any missing Spaces
+        for (int zoneNum = 1; zoneNum <= state.dataGlobal->NumOfZones; ++zoneNum) {
+            auto &thisZone = state.dataHeatBal->Zone(zoneNum);
+            if (thisZone.anySurfacesWithoutSpace) {
+                // If any surfaces in the zone are not assigned to a space, may need to create a new space
+                // Every zone has at least one space, created in HeatBalanceManager::GetSpaceData
+                // If no surfaces have a space assigned, then the default space will be used, otherwise, create a new space
+                if (thisZone.anySurfacesWithSpace) {
+                    // Add new space
+                    ++state.dataGlobal->numSpaces;
+                    state.dataHeatBal->space(state.dataGlobal->numSpaces).zoneNum = zoneNum;
+                    // Add to zone's list of spaces
+                    thisZone.spaceIndexes.emplace_back(state.dataGlobal->numSpaces);
+                    ++state.dataHeatBal->Zone(zoneNum).numSpaces;
+                    assert(state.dataHeatBal->Zone(zoneNum).numSpaces == int(state.dataHeatBal->Zone(zoneNum).spaceIndexes.size()));
+                    // If some surfaces in the zone are assigned to a space, the new space is the remainder of the zone
+                    state.dataHeatBal->space(state.dataGlobal->numSpaces).Name = thisZone.Name + "-Remainder";
```

Comment:
This is where the "-Remainder" spaces get created if needed. The full-zone default spaces were created in `HeatBalanceManager::GetSpaceData`.